www/js/metrics/metricsHelper.ts

```diff
@@ -0,0 +1,212 @@
+  const dt = DateTime.fromISO(day.fmt_time, { zone: 'utc' });
+  return dt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+}
+
+export function formatDateRangeOfDays(days: DayOfMetricData[]) {
+  if (!days?.length) return '';
+  const firstDayDt = DateTime.fromISO(days[0].fmt_time, { zone: 'utc' });
+  const lastDayDt = DateTime.fromISO(days[days.length - 1].fmt_time, { zone: 'utc' });
+  const firstDay = firstDayDt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+  const lastDay = lastDayDt.toLocaleString({...DateTime.DATE_SHORT, year: undefined});
+  return `${firstDay} - ${lastDay}`;
+}
+
+/* formatting data form carbon footprint calculations */
+
+//modes considered on foot for carbon calculation, expandable as needed
+const ON_FOOT_MODES = ['WALKING', 'RUNNING', 'ON_FOOT'] as const;
+
+/*
+* metric2val is a function that takes a metric entry and a field and returns
+* the appropriate value.
+* for regular data (user-specific), this will return the field value
+* for avg data (aggregate), this will return the field value/nUsers
+*/
+const metricToValue = function(population:'user'|'aggreagte', metric, field) {
+  if(population == "user"){
+    return metric[field];
+  }
+  else{
+    return metric[field]/metric.nUsers;
+  }
+}
+
+//testing agains global list of what is "on foot"
+//returns true | false
+const isOnFoot = function(mode: string) {
+  for (let ped_mode in ON_FOOT_MODES) {
+    if (mode === ped_mode) {
+      return true;
+    }
+  }
+  return false;
+}
+
+//from two weeks fo low and high values, calculates low and high change
+export function calculatePercentChange(pastWeekRange, previousWeekRange) {
+  let greaterLesserPct = {
+    low: (pastWeekRange.low/previousWeekRange.low) * 100 - 100,
+    high: (pastWeekRange.high/previousWeekRange.high) * 100 - 100,
+  }
+  return greaterLesserPct;
+}
+
+export function parseDataFromMetrics(metrics, population) {
+  console.log("Called parseDataFromMetrics on ", metrics);
+  let mode_bins = {};
+  metrics?.forEach(function(metric) {
+    let onFootVal = 0;
+
+    for (let field in metric) {
+      /*For modes inferred from sensor data, we check if the string is all upper case 
+      by converting it to upper case and seeing if it is changed*/
+      if(field == field.toUpperCase()) {
+        /*sum all possible on foot modes: see https://github.com/e-mission/e-mission-docs/issues/422 */
+        if (isOnFoot(field)) {
+          onFootVal += metricToValue(population, metric, field);
+          field = 'ON_FOOT';
+        }
+        if (!(field in mode_bins)) {
+          mode_bins[field] = [];
+        }
+        //for all except onFoot, add to bin - could discover mult onFoot modes
+        if (field != "ON_FOOT") {
+          mode_bins[field].push([metric.ts, metricToValue(population, metric, field), metric.fmt_time]);
+        }
+      }
+      //this section handles user lables, assuming 'label_' prefix
+      if(field.startsWith('label_')) {
```

Comment:
we should be able to split the parsing of inferred versus user label modes into two separate functions instead of this giant function - e.g. the two `if` statements

Reply:
TODO cleanup